### ISSUE TITLE
Adapt the default logging scheme of Lightning

### ISSUE THREAD

**Issue by adosar** [adosar/aidsorb#50]
Currently:
```python
self.train_metric.update(...)
self.log_dict(self.train_metric, on_step=False, on_epoch=True, prog_bar=True)
```
The step values to be reduced are obtained with different parameters values (i.e. we don't fix the parameters and then make a full pass of the training set). As such, this "average" metric at the end of the epochs feels like a "smoothing" average of previous steps. 

Logging training metrics on step-level is another alternative (Lightning's default logging scheme). Still, train metric isn't computed over the whole dataset, however it is closer to the "current" training performance on would expect during training.

**Adapting to this scheme, will also make it easier for the users to remember what is going on under the hood**.

Suggested change:
```python
self.train_metric(...)  # Remove update
self.log_dict(self.train_metric, prog_bar=True)  # Lightning uses on_step=True and on_step=False for this hook.
```
